src/celldega/sp/__init__.py

```diff
@@ -0,0 +1,247 @@
+"""
+Module for spatial operation and analysis
+"""
+
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+import matplotlib.cm as cm
+import geopandas as gpd
+import pandas as pd
+import colorsys
+import numpy as np
+import json
+import celldega as dega
+
+def calc_distance_to_roi(
+    gdf_polygons: gpd.GeoDataFrame,
+    gdf_points: gpd.GeoDataFrame,
+    roi_name: str
+) -> tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """
+    Calculate the distance of points to the specified region of interest.
+    
+    Args:
+        gdf_polygons: GeoDataFrame containing polygons with a 'roi' column.
+        gdf_points: GeoDataFrame containing points to calculate distances for.
+        roi_name: Name of the region of interest (ROI) to calculate distances to.
+    
+    Returns:
+        A tuple containing:
+            - gdf_points: The input gdf_points with added distance column.
+            - gdf_polygon: The extracted polygon for the specified ROI.
+    """
+
+    print (f"Extract the polygon for the specified ROI: {roi_name}")
+    # Extract the polygon for the specified ROI
+    gdf_polygon = gdf_polygons.loc[gdf_polygons["roi"] == roi_name].reset_index()
+    
+    # Calculate distances from points to the polygon
+    polygon_geom = gdf_polygon.geometry.iloc[0]
+
+    gdf_points_with_dist = gdf_points.copy()
+    gdf_points_with_dist[f"distance_to_polygon_{roi_name}"] = gdf_points_with_dist.distance(polygon_geom)
+    print (f"Distances from points to polygon {roi_name} were calculated.")
+    return gdf_points_with_dist, gdf_polygon
+
+def plot_distance_to_roi(
```

Comment:
I think going forward we can try to put all plotting methods under the viz module, but we can rearrange this later if needed.

Reply:
fixed